interfaces/apparmor/backend.go

```diff
@@ -63,6 +70,107 @@ func (b *Backend) Name() interfaces.SecuritySystem {
 	return interfaces.SecurityAppArmor
 }
 
+// Initialize prepares customized apparmor policy for snap-confine.
+func (b *Backend) Initialize() error {
+	// TODO: also generate policy for snap-confine from core.
+	return setupSnapConfineGeneratedPolicy()
+}
+
+// setupSnapConfineGeneratedPolicyImpl inspects the system and sets up local
+// apparmor policy for snap-confine. Local policy is included by the
+// system-wide policy. If the local policy has changed then the apparmor
+// profile for snap-confine is reloaded.
+func setupSnapConfineGeneratedPolicyImpl() error {
+	// Location of the generated policy.
+	dir := dirs.SnapConfineAppArmorDir
+	glob := "generated-*"
+	policy := make(map[string]*osutil.FileState)
+
+	// Create the local policy directory if it is not there.
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("cannot create snap-confine policy directory, %s", err)
+	}
+
+	// Check if NFS is mounted at or under $HOME. Because NFS is not
+	// transparent to apparmor we must alter our profile to counter that and
+	// allow snap-confine to work.
+	if nfs, err := isHomeUsingNFS(); err != nil {
```

Comment:
I wonder if an inability to determine if we are using NFS here should be fatal (it's fatal in this code if I'm not mistaken), unless we've full faith in our fstab/mountinfo parsing (which we probably need to have given that it's used in some other places afair). Anyway, just mentioning the concern since NFS is not a typical use case - perhaps a warning would be enough. For your consideration.

Reply:
Interesting question. While we don't expect any errors I agree we should stop the world about it. It is just a workaround for a missing kernel feature (NFS transparency in LSMs). I'll make this non fatal.

Reply:
Done now.

Reply:
Thank you!